### ISSUE TITLE
Make a context manager for save() / restore()

### ISSUE THREAD

**Issue by justvanrossum** [typemytype/drawbot#140]
See discussion in #98.

It will be easy to do, but we need to decide on a name. We could even use `save()` for this without losing b/w compatibility:

```python
with save():
    rotate(45)
    rect(0, 0, 100, 100)
```

But maybe `savedGState()` would be better, because it's more explicit.

**Comment by typemytype** [comment]
maybe just keep `save` :)

```pytyhon
import drawBot

class SaveContextManager(object):
    
    def __call__(self):
        print "call"
        drawBot.save()
        return self
    
    def __enter__(self):
        print "enter"
        drawBot.save()
        return self

    def __exit__(self, type, value, traceback):
        print "exit"
        drawBot.restore()

# return this from drawingTools        
save = SaveContextManager()


with save:
    print "ll"

save()
```

**Comment by justvanrossum** [comment]
Hmm, imagine if someone accidentally does

```python
with save():
  ...
```

Would it be bad is `drawBot.save()` is done in `__init__`, and `__enter__` just returns self without doing anything else?

**Comment by justvanrossum** [comment]
In other words: syntactically, I'd rather see `with save(): ...` than `with save: ...`

**Comment by typemytype** [comment]
true, offcourse it just could return self on __enter__

**Comment by justvanrossum** [comment]
Closed by merging #149.
